server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchDocValuesContext.java

```diff
@@ -23,20 +24,26 @@
  */
 public class FetchDocValuesContext {
 
-    private final List<FieldAndFormat> fields = new ArrayList<>();
+    private final Collection<FieldAndFormat> fields;
 
     /**
      * Create a new FetchDocValuesContext using the provided input list.
      * Field patterns containing wildcards are resolved and unmapped fields are filtered out.
      */
     public FetchDocValuesContext(SearchExecutionContext searchExecutionContext, List<FieldAndFormat> fieldPatterns) {
+        // Use Linked HashMap to reserve the fetching order
+        final Map<String, FieldAndFormat> fieldToFormats = new LinkedHashMap<>();
         for (FieldAndFormat field : fieldPatterns) {
             Collection<String> fieldNames = searchExecutionContext.getMatchingFieldNames(field.field);
             for (String fieldName : fieldNames) {
-                fields.add(new FieldAndFormat(fieldName, field.format, field.includeUnmapped));
+                final FieldAndFormat format = new FieldAndFormat(fieldName, field.format, field.includeUnmapped);
+                final FieldAndFormat existing = fieldToFormats.put(fieldName, format);
+                if (existing != null && existing.equals(format) == false) {
```

Comment:
I wonder if it'd be better to just have the first definition take precedence, instead of throwing an error. Being flexible here seems less disruptive, and it matches what we do for the `fields` option. I also don't think this behavior is too "trappy" for users.

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/89094/commits/63a3f25be1822dfd68a648abc3fa4020dcdd18c9 to remove the restriction.